exporter/awsemfexporter/internal/useragent/useragent.go

```diff
@@ -107,14 +119,17 @@ func (ua *UserAgent) ProcessMetrics(metrics pmetric.Metrics) {
 			ms := ilms.At(j).Metrics()
 			for k := 0; k < ms.Len(); k++ {
 				metric := ms.At(k)
-				if strings.HasPrefix(metric.Name(), ebsMetricPrefix) {
-					ua.featureList[attributeEBS] = struct{}{}
-					ua.build()
-					return
+				for feature, prefix := range featureMetricPrefixes {
+					if strings.HasPrefix(metric.Name(), prefix) {
+						if _, exists := ua.featureList[feature]; !exists {
+							ua.featureList[feature] = struct{}{}
+						}
+					}
 				}
 			}
 		}
 	}
+	ua.build()
```

Comment:
nit: Instead of always rebuilding this, add a `featureFound` bool that you can set to true only when a feature is truly found so you know when to rebuild.